### ISSUE TITLE
Variant agnostic build steps

### ISSUE THREAD

**Issue by ttanimura** [nerdvegas/rez#394]
Is there an existing way of executing project-level (i.e. variant agnostic) build steps for packages with variants? Our packages contain a target to build Sphinx docs which are the same across all variants, so ideally we would like to only build them once. 

Currently Rez treats each variant as a separate cmake project, so `PROJECT_BINARY_DIR` points to the variant's root directory, not the project's. I had a look through the cmake env vars and variables, but couldn't find anything that represented the project-level build directory. I can work around this with some path hackery, but it seems error-prone.

Thanks!

**Comment by nerdvegas** [comment]
There's no way to do this currently. Even if there were I'm not sure how it
should work. Perhaps introduce a 'base' build? But that would surely break
existing builds. Unless it's opt-in on a per-package+configurable basis.

Will have to think about but this is something that's come up a number of
times.

A

**Comment by ttanimura** [comment]
Yeah, its not something that makes sense with the way Rez works. What about a post-build type of thing? I guess it would be at the package scope, although perhaps that is something that can be configurable.

**Comment by maxnbk** [comment]
I've had cases where I needed builds like this. ie, I'd like to generate variants of packages, but I only need to build a payload which those packages point to just once, because for example, a payload might come with libraries for multiple versions of Nuke already bundled, and it doesn't make much sense to "build" it multiple times.
I tend to get around it by not using variants and instead introducing a requirement for the variant range, and letting the commands() section handle any different pathing needed. It's not ideal but fits the concept reasonably well (of having a "base" build)
